### DISCUSSION TITLE
colour speed on tracks

### DISCUSSION THREAD

**Discussion by dizmoduck** [OpenTracksApp/OpenTracks#1505] [Ideas]
it will be good to be able to see one's differentiated speed on the map
Fx
green=faster
yellow =medium
red=slower

**Comment by pstorch** [comment]
Interesting idea. Would need to be configurable what faster, medium and slower means exactly. Maybe depending on the type of activity.

**Reply by dizmoduck** [reply]
I think more about something general, e.g. percentage above/below medium speed - KISS principle😁

Medium is avg speet/pace
and if it varies more than, for example, ±10%, it will be red or green, the % can be configurable

Btw GeoTacker has It, which I have used for many years, but a lack of development and statistics got me over to OpenTacker 😁

**Reply by dennisguse** [reply]
I like the idea with deviation from the average. 
One question is then how we handle changes: I guess they should not be to abrupt, right?

And how do we handle if the average changes (i.e., while still recording)?

**Reply by dizmoduck** [reply]
I don't think that it should be done while it is being recorded, far too advanced, but only when tracking is finished and you want to view it on the map

**Comment by pstorch** [comment]
I've created a Ticket for OSMDashboard: https://github.com/OpenTracksApp/OSMDashboard/issues/205

**Comment by EliasA** [comment]
Would it be possible to make these colours configurable? Reason for this: colour combination red - yellow - green is uninformative for colour blind persons who cannot see a difference between red and green. 8,5 % of the population is colour blind - 8 % of males and 0,5 % of females.

**Comment by pstorch** [comment]
@EliasA thanks for your suggestion. I would love to make it more accessible. Can you describe what colors would be best and how to apply them to the speed value?

**Reply by EliasA** [reply]
Wow - this was fast! I am not colour blind myself so I will ask for opinion person's opininion on this. I'll ask around in Mastodon so the answers will be easy to track. 

At the same time we can also make the colours contrastwise as good as possible even for people who are not visually impaired. I am not at all convinced that the present colours are clearly visible if user has a dark map theme - even though I do not know whether it is possible or not. As dark themes are in fashion I think there's a reason for some thinking here. 

I have some experience in developing the visual presentation of electronic marine charts in night mode. I'll try to add my five cents to the development. Thank you for the flying start! ;-)

What are the criteria for showing or the legend for different colours at the present?

**Reply by pstorch** [reply]
The color by speed is quite easy right now. It uses the RGB color value. Start by full red, reduce to 0 red and then increase green until full. If we let the user choose two arbitrary colors I need to find out how to blend them into each other.

For the color by track mode it would be easier. Currently there is a color calculator (I borrowed from somewhere) which gives me a new color each time I ask for it. I think this could be replaced by configurable 5-10 fixed colors. If more tracks are shown on a map it could repeat the colors. I guess it is a rare case anyhow.

**Reply by dennisguse** [reply]
IIRC https://github.com/dennisguse/TheKarte/blob/main/src%2FStyleColorCreator.js

@pstorch

**Reply by pstorch** [reply]
Exactly, @dennisguse, that was the source of the color creator. 😎

**Reply by pstorch** [reply]
@EliasA I made the colors configurable. Can be tested in the next nightly build.

**Reply by EliasA** [reply]
@pstorch Thanks - I just installed version 6108 and version 1440 of the OSM plugin. I can see the colours are now different but I cannot find where the configuration is done. Where can I find the settings?

**Reply by pstorch** [reply]
The cog wheel:

![Screenshot_20250317_171206_OSM Dashboard (Debug)](https://github.com/user-attachments/assets/f37db5af-c663-4c84-9720-3eca0c4e0230)

**Reply by EliasA** [reply]
I feel stupid. I guessed it as in the menu that is opened with the cog wheel but I still can't find a menu entry for configuring colours. I am able to find Dynamic Colors slide and Track color entry with three options.

Have I understood something wrong here? Does "configurable colours" mean user can define what colours are used or choosing between the three options?

**Reply by pstorch** [reply]
Sorry, my explanation was too short.
Can you see these color settings, via the cog wheel and scrolling down a bit?

There are 5 different colors for the color by track mode and 3 colors for the color by speed mode (for low, average and high speed). 

![Screenshot_20250317_220439_OSM Dashboard (Debug)](https://github.com/user-attachments/assets/1143ed6b-59b8-466d-ab2e-8a62a415ef67)

**Reply by EliasA** [reply]
No I cannot see entry for colors at all. There is one odd thing that might have something to do with it: I have the user interface in Finnish. In spite of that the name of the menu (where in your picture there is text Settings), says Inställningar which is swedish for Settings. 

I of course can change the language of UI temporarily if that helps in debugging.

**Reply by pstorch** [reply]
That's strange. Can you post a screenshot?
And yes, please try a different language (English default), if that helps.

**Reply by EliasA** [reply]
After I rebooted my phone the menu looks like this - the name of the menu is Settings - in English. :O

![kuva](https://github.com/user-attachments/assets/fd8d3c06-b5c5-48c2-bded-b44d95ec3843)

**Reply by pstorch** [reply]
Good so far, can you scroll down?

**Reply by EliasA** [reply]
Yes, I can. After changing the system language to English (Finnish locale though) and the OSM plugin language to English (United States) I see the following. Sorry - I don't have a tool for screenshots longer than the screen... (Is there a F-Droid app for that?)

![kuva](https://github.com/user-attachments/assets/ddddcac2-79b8-4cb6-87a2-f1e74968426f)
![kuva](https://github.com/user-attachments/assets/f27a0385-3af7-4f20-a055-96695f5bdc60)
![kuva](https://github.com/user-attachments/assets/f6c27afd-6976-460c-a142-b59e05ad643b)

**Reply by pstorch** [reply]
Oh, that is really strange.
The color settings should be between Map theme directory and the advanced settings,like here:

![Screenshot_20250317_224746_OSM Dashboard (Nightly)](https://github.com/user-attachments/assets/b820a57a-6954-42cc-8285-b46dd913dab4)

Don't know what is wrong here. Maybe I can try different android versions in the emulator. But that would need some time.
What android version do you have?

**Reply by EliasA** [reply]
I am using Android 15, LineageOS 22.1.

**Reply by dennisguse** [reply]
@pstorch for me everything works using the nightly build.

**Reply by pstorch** [reply]
I tried with 3 different emulated devices with Andorid 14 and 15. All show the color preferences. Don't know what else to try. 
I don't have a phone with LineageOS at hand.

**Reply by dennisguse** [reply]
@EliasA which version of OSM Dashboard are you using?

**Reply by EliasA** [reply]
I downloaded and installed 1440 nightly of OSM Dashboard at the same time as  I downloaded and installed OpenTracks 6108 nightly.

**Reply by EliasA** [reply]
I'll do some testing with older spare phones running something else than LOS.

**Reply by pstorch** [reply]
@EliasA just some guesses: was it a fresh install? Have you tried to uninstall/reinstall the nightly build? Do you have the release version installed in parallel? Just to make sure it's not mixed up. I've no good idea that's why I'm grasping at straws.

**Reply by EliasA** [reply]
Initially I had release version installed when I installed 6108 nightly + OSM Dashboard 1440. When I noticed there was a problem I uninstalled the release version. No difference. I just uninstalled the test versions of OpenTracks and OSM Dashboard and reinstalled them. No difference.

**Reply by pstorch** [reply]
Thanks for trying.

**Reply by pstorch** [reply]
@EliasA have you had the chance to test with another phone?

**Reply by EliasA** [reply]
Sorry - not yet. Thanks for your reminder - I'll try to do it tonight.
